Pandas/CSV/TSV as Observer/Source supporting dataframe, CSV, and Excel

cnarte opened the pull request:
This fork helps to parse data from pandas dataframe or from user input of a csv file or excel sheets.
The input_type of target needs to be specified from this list. ["csv","dataframe","excel'].
Added example script too for testing purposes in example dir.
Resolves #169

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/obsei/obsei?pullRequest=184) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/obsei/obsei?pullRequest=184) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/obsei/obsei?pullRequest=184) it.</sub>

lalitpagaria commented on example/pandas_scrapper.py:
```diff
@@ -0,0 +1,20 @@
+from obsei.source.pandas_source import (
```
let's change the name of file. As this is not a scrapper.

cnarte replied:
Sure sir I'll name this script pandas_source_example.py? That better ?

lalitpagaria replied:
Yes this would be better name

lalitpagaria commented on obsei/source/pandas_source.py:
```diff
@@ -0,0 +1,103 @@
+
+
+class PandasSourceConfig(BaseSourceConfig):
+    TYPE: str = "PandasScrapper"
```
Correct the name it should be "Pandas" only

cnarte replied:
Okay sir, I'll change it to pandas.

lalitpagaria commented on pandas_scrapper.py:
```diff
@@ -0,0 +1,21 @@
+from obsei.source.pandas_source import (
```
What is this file do?

lalitpagaria replied:
@cnarte please review this file

cnarte replied:
Resolved this one sir now the file is named Pandas only 
![image](https://user-images.githubusercontent.com/56212417/131347734-f436fd4e-6989-4aad-bf36-b04905de3783.png)

lalitpagaria replied:
Means you already have [this ](https://github.com/obsei/obsei/pull/184/files#diff-1ec87da45258e6613f4ca964bf22b3264a616bf6f5f6e49ddbd063fa67f11b5b) then why you need this file

lalitpagaria commented on obsei/source/pandas_source.py:
```diff
@@ -0,0 +1,103 @@
+logger = logging.getLogger(__name__)
+
+
+class PandasSourceConfig(BaseSourceConfig):
```
Let's keep it simple and remove code related CSV/Excel.
CSV/Excel need different way of handling. 
Like reading -
- Very large file (lets say in TBs)
- Reading file from local, and remote
- Reading continuously updating file (ie CSV, Excel etc file are getting updated regularly while reading and there is need of code to handle that)

cnarte replied:
Sure sir ,
- I'll comment the current csv and excel related code .
- For any data bigger than 1GB we would need to use something like pyspark or any other big-data platform that supports in memory computations.

lalitpagaria replied:
Yes exactly, future plan to incorporate it. means providing full production grade setup (data streaming and cloud hosting setup)

lalitpagaria commented on obsei/source/pandas_source.py:
```diff
@@ -0,0 +1,103 @@
+class PandasSourceConfig(BaseSourceConfig):
+    TYPE: str = "PandasScrapper"
+    target: str = None
+    target_df: Optional[pd.DataFrame] = None
```
Let's keep only three variables PandasSinkConfig -
```
class PandasSourceConfig(BaseSinkConfig):
    TYPE: str = "Pandas"
    dataframe: Optional[DataFrame] = None
    # By default it will add all fields to meta
    include_columns_list: Optional[List[str]] = None
    # Text columns
    text_columns_list: List[str]
```

And constructor need to perform validation about text_columns_list

cnarte replied:
I'm refactoring the code for only dataframe. I'll also include required validations.

lalitpagaria commented on obsei/source/pandas_source.py:
```diff
@@ -0,0 +1,103 @@
+    ) -> List[TextPayload]:
+        source_responses: List[TextPayload] = []
+
+        # Get data from state
```
I think pandas dataframe can't have any state information. But we can keep last execution time information.

lalitpagaria commented on obsei/source/pandas_source.py:
```diff
@@ -0,0 +1,103 @@
+
+        non_target_col_list = [x for x in df.columns if x not in config.columns]
+        for col in config.columns:
+            for index, row in df.iterrows():
```
Please check time complexity of these two loops.
For example what will happen when we have 10000000 rows and 1000 columns.

cnarte replied:
As now only target is a dataframe I'll be using df.apply which is faster cython based immplementaion.

lalitpagaria reviewed (changes requested):
@cnarte I have reviewed PR. Have comments please address them. Let's keep first thing simple.

lalitpagaria commented:
Also check type check errors as well

cnarte commented:
Made changes as per recommended :

-  renamed pandas_scrapper ---> pandas_source_example
- renamed pandas_source ---> pandas
- Removed csv and excel code
- Now df.apply is being used for parsing purpose as its based on cython and faster
